UnitySDK/Assets/ML-Agents/Scripts/Agent.cs

```diff
@@ -292,6 +292,11 @@ void NotifyAgentDone()
             NNModel model,
             InferenceDevice inferenceDevice = InferenceDevice.CPU)
         {
+            if (m_PolicyFactory == null)
+            {
+                // TODO better way to do this lazily
+                m_PolicyFactory = GetComponent<BehaviorParameters>();
```

Comment:
At least in my one test, the ModelNotifier.OnEnable was called before the Agent, so m_PolicyFactory wasn't initialized yet. Is there a better way to go about this?

Reply:
maybe the initialization of the `BehaviorParameters` (and other 'static' members) should be moved to `Awake`

Reply:
Maybe we should create an AgentLazyInitialize to make sure it is initialized correctly?
This way we could raise proper errors if the m_PolicyFactory is not attached.

Reply:
Actually, I think I can get around this for now by accessing the BehaviorParameters directly.

Reply:
Ended up doing Agent.LazyInitialize(). This makes some unit test code cleaner too.